tests/test_time.py

```diff
@@ -5,20 +5,129 @@
 import pytest
 import xarray as xr
 from s2spy.time import AdventCalendar
+from s2spy.time import CustomCalendar
 from s2spy.time import MonthlyCalendar
+from s2spy.time import Period
+from s2spy.time import PrecursorPeriod
+from s2spy.time import TargetPeriod
 from s2spy.time import WeeklyCalendar
 
 
-def interval(start, end):
+def interval(start, end, closed="right"):
     """Shorthand for more readable tests."""
-    return pd.Interval(pd.Timestamp(start), pd.Timestamp(end))
+    return pd.Interval(pd.Timestamp(start), pd.Timestamp(end), closed = closed)
+
+
+class TestPeriod:
+    """Test Period objects."""
+
+    def test_period(self):
+        period = Period(length=10, gap=10)
+        assert isinstance(period, Period)
+
+    def test_target_period(self):
+        target = TargetPeriod(20, 10)
+        assert isinstance(target, TargetPeriod)
+
+    def test_precursor_period(self):
+        precursor = PrecursorPeriod(10, -5)
+        assert isinstance(precursor, PrecursorPeriod)
+
+
+class TestCustomCalendar:
+    """Test CustomCalendar methods."""
+
+    @pytest.fixture(autouse=True)
+    def dummy_calendar(self):
+        cal = CustomCalendar(anchor="12-31")
```

Comment:
We should probably put in a test that tests a set of anchor inputs

Reply:
Thanks for the comment. Let's create a separate test (specifically for the new parser) in another PR for the different type of anchor inputs.